Bottom: fd861e94d3db0809741011a50b00c8807a1ab7d8
Top:    bf2d8679f83bedf1d3bddbe7bde80d49089dd372
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-03 14:49:08 +0000

polygon1.c: Temporarily disable the seg_in_region test

The code does not handle the bounding region of arcs yet


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 78ba618..9157eb4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -600,6 +600,7 @@ adjust_tree (rtree_t * tree, struct seg *s)
   return 0;
 }
 
+#if 0  /* TURNS OUT DISABLING THIS ACTUALLY (VERY MARGINALLY) SPEEDS THINGS UP! */
 /*
  * seg_in_region()
  * (C) 2006, harry eaton
@@ -621,6 +622,7 @@ seg_in_region (const BoxType * b, void *cl)
     return 0;
   return 1;			/* might intersect */
 }
+#endif
 
 /* Prepend a deferred node-insersion task to a list */
 static insert_node_task *
@@ -844,7 +846,7 @@ contour_bounds_touch (const BoxType * b, void *cl)
       info.tree = rtree_over->tree;
       if (info.tree)
 	if (UNLIKELY (r_search (info.tree, &info.s->box,
-				seg_in_region, seg_in_seg, &info)))
+				NULL /* seg_in_region */, seg_in_seg, &info)))
 	  assert (0); /* XXX: Memory allocation failure */
     }
   while ((av = av->next) != &looping_over->head);
